### PR TITLE
ATDM: Upgrade to CMake 3.17+ and fix intel compiler checks 

### PR DESCRIPTION
**Pull request by e10harvey**
Related to #8675.

## Sems-rhel7 testing in progress:
https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2021-02-03&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=sems-rhel7&field2=groupname&compare2=63&value2=Experimental

## cts1empire testing in progress:
https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2021-02-03&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=cts1empire&field2=groupname&compare2=63&value2=Experimental

## van1-tx2 testing in progress:
https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2021-02-03&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=van1-tx2&field2=groupname&compare2=63&value2=Experimental

### CODE REVIEW

**Comment by jmgate** [comment]
Do we want to go to 3.19 everywhere if it's available?  I thought I'd seen a PR updating `cts1empire` already&mdash;am I mistaken there?

**Comment by e10harvey** [comment]
> Do we want to go to 3.19 everywhere if it's available?

I don't have a preference. cmake 3.19 is not available on van1-tx2.




> I thought I'd seen a PR updating `cts1empire` already—am I mistaken there?

I do not see a PR updating cmake for `cts1empire1`. @jmgate: do you have a link?

**Review by jmgate — APPROVED** [review]
Looks good.  Feel free to merge.

File: cmake/std/atdm/cee-rhel7/environment.sh [hunk 1]
```diff
@@ -262,5 +262,8 @@ if [[ "${ATDM_CONFIG_SUPERLUDIST_INCLUDE_DIRS}" == "" ]] ; then
   export ATDM_CONFIG_SUPERLUDIST_LIBS=${SUPERLUDIST_ROOT}/lib64/libsuperlu_dist.a
 fi
 
+# Point CMake 3.19 compiler checks to missing symbols
+export LDFLAGS="$LDFLAGS -lifcore"
```
Comment (jwillenbring) [inline cmake/std/atdm/cee-rhel7/environment.sh:266]: This is only needed for intel builds. Other builds it causes an error as this library is not available in that environment. This one for example is clang:

https://testing.sandia.gov/cdash/build/8654297/configure

This one is GCC:

https://testing.sandia.gov/cdash/build/8654262/configure

-lifcore needs to be added individually to the 3 builds that were failing yesterday only.

**Pull request merged**